### PR TITLE
`node(s)` return native coordinates for grids + `x/y/znode(s)` -> distances & `λ/φnode(s)` -> degrees

### LINKED ISSUE #2978: `x/y/znode` returns different things for different grids
There is an issue that `xnode` returns distances for `RectilinearGrid` but returns degrees for `LatitudeLongitudeGrid`.

We run onto this with @glwagner while working on horizontal regrading in #2881.

One suggestion is that `node()` and `nodes()` are defined for each grid and that it returns that native grid coordinates, e.g.

```julia
nodes(grid::RectilinearGrid)       = (xnodes(grid), ynodes(grid), znodes(grid))
nodes(grid::LatitudeLongitudeGrid) = (λnodes(grid), φnodes(grid), znodes(grid))
```

Then within grids we define the `xnode`, `xnodes` but also the native-coordinates-analogues, e.g., `λnode`, `λnodes`.

### PR DESCRIPTION
**Pull request by navidcy**
This PR makes `node` and `nodes` return the native coordinates of the grid. It changes the behavior of `x/ynode(s)`. Now `x/ynode(s)` _always_  returns `x/y` distances. The PR also introduces `λ/φnode(s)` for curvilinear grids.

This way, now `node(s)` behaviour didn't change. But now they return:

- `(xnode(s), ynode(s), znode(s))` for RectilinearGrid
- `(λnode(s), φnode(s), znode(s))` for CurvilinearGrids

Note that still there are `x/y/znode(s)` methods _for all grids_ and those methods actually return distances (in meters). Before this PR when `ynode` called on, e.g., a LatitudeLongitude grid would return latitude in degrees and this was tad confusing.

Closes #2978

### CODE REVIEW

**Comment by glwagner** [comment]
Might want to hold off until multi region stuff is merged cause I'm making a huge overhaul to everything grids there

**Comment by navidcy** [comment]
Yes, this is a placeholder :)

**Comment by navidcy** [comment]
OK, now we should resume working on this ;)

**Comment by navidcy** [comment]
@glwagner can you fix
https://github.com/CliMA/Oceananigans.jl/blob/f1c1a907cf9bd176493188a8e0966f2344c4f0d8/src/Grids/grid_utils.jl#L44
as you were suggesting in #2991?

**Comment by navidcy** [comment]
@simone-silvestri can you review this?

**Comment by navidcy** [comment]
@simone-silvestri, in relation to why `cpu_face_constructors` require `with_halos=true`: note that before the grid_utils refactor `cpu_face_constructors` were using `cpu_face_constructors` were using `all_x/y/z_nodes` which is exactly equivalent to `x/y/znodes(; with_halos=true)`. ;)

https://github.com/CliMA/Oceananigans.jl/blob/e7cb507b2645f04c6cdf52ee6f18c2c5b79f7bde/src/Grids/grid_utils.jl#L244-L246

File: src/Grids/latitude_longitude_grid.jl [hunk 1]
```diff
@@ -584,31 +584,79 @@ return_metrics(::LatitudeLongitudeGrid) = (:λᶠᵃᵃ, :λᶜᵃᵃ, :φᵃᶠ
...
-
-@inline ynodes(grid::LatLonGrid, ℓy::Face  ; with_halos=false) = with_halos ? grid.φᵃᶠᵃ : view(grid.φᵃᶠᵃ, interior_indices(ℓy, topology(grid, 2)(), size(grid, 2)))
-@inline ynodes(grid::LatLonGrid, ℓy::Center; with_halos=false) = with_halos ? grid.φᵃᶜᵃ : view(grid.φᵃᶜᵃ, interior_indices(ℓy, topology(grid, 2)(), size(grid, 2)))
+function nodes(grid::LatitudeLongitudeGrid, ℓx, ℓy, ℓz; reshape=false, with_halos=false)
+    λ = λnodes(grid, ℓx, ℓy, ℓz; with_halos)
+    φ = φnodes(grid, ℓx, ℓy, ℓz; with_halos)
+    z = znodes(grid, ℓx, ℓy, ℓz; with_halos)
+
+    if reshape
+        N = (length(x), length(y), length(z))
```
Comment (glwagner) [inline src/Grids/latitude_longitude_grid.jl:593]:
```suggestion
        N = (length(λ), length(φ), length(z))
```

File: src/Grids/orthogonal_spherical_shell_grid.jl [hunk 2]
```diff
@@ -864,47 +864,89 @@ end
...
 
-@inline xnodes(grid::OSSG, ℓx::Face,   ℓy::Face, ; with_halos=false) = with_halos ? grid.λᶠᶠᵃ :
+function nodes(grid::OSSG, ℓx, ℓy, ℓz; reshape=false, with_halos=false)
+    λ = λnodes(grid, ℓx, ℓy, ℓz; with_halos)
+    φ = φnodes(grid, ℓx, ℓy, ℓz; with_halos)
+    z = znodes(grid, ℓx, ℓy, ℓz; with_halos)
+
+    if reshape
+        # x and y are 2D arrays
+        N = (size(x)..., size(z)...)
```
Comment (glwagner) [inline src/Grids/orthogonal_spherical_shell_grid.jl:874]:
```suggestion
        N = (size(λ)..., size(z)...)
```

**Review by glwagner — APPROVED** [review]

**Pull request merged**